packages/block-editor/src/components/block-list/index.js

```diff
@@ -218,6 +223,90 @@ function Items( {
 			[ rootClientId, hasAppender, hasCustomAppender ]
 		);
 
+	const {
+		isZoomOut,
+		sectionRootClientId,
+		sectionClientIds,
+		blockInsertionPoint,
+		blockInsertionPointVisible,
+	} = useSelect( ( select ) => {
+		const {
+			getSettings,
+			__unstableGetEditorMode,
+			getBlockInsertionPoint,
+			getBlockOrder,
+			isBlockInsertionPointVisible,
+		} = unlock( select( blockEditorStore ) );
+
+		if ( __unstableGetEditorMode() === 'zoom-out' ) {
+			const { sectionRootClientId: root } = unlock( getSettings() );
+			const sectionRootClientIds = getBlockOrder( root );
+			return {
+				isZoomOut: __unstableGetEditorMode() === 'zoom-out',
+				sectionRootClientId: root,
+				sectionClientIds: sectionRootClientIds,
+				blockOrder: getBlockOrder( root ),
+				blockInsertionPoint: getBlockInsertionPoint(),
+				blockInsertionPointVisible: isBlockInsertionPointVisible(),
+			};
+		}
+		return {};
+	}, [] );
+
+	const isReducedMotion = useReducedMotion();
+	const renderZoomOutSeparator = ( clientId, position = 'top' ) => {
```

Comment:
Can we extract this as a separate component? It can be in the same file. Similar render functions are usually hard to maintain.

Reply:
Good idea. I moved it to a different file as well, as this one is really big.